browser/sources.gni

```diff
@@ -582,19 +581,17 @@ brave_chrome_browser_deps += brave_browser_web_discovery_deps
 brave_chrome_browser_deps += brave_chromium_src_chrome_browser_deps
 brave_chrome_browser_deps += brave_chromium_src_chrome_browser_prefs_deps
 
-brave_chrome_browser_allow_circular_includes_from = [ "//brave/browser/ui" ]
+brave_chrome_browser_allow_circular_includes_from = [
+  "//brave/browser/ui",
+  "//brave/browser/brave_ads:impl",
```

Comment:
this allows `//brave/browser/brave_ads:impl` to use `//chrome/browser` dependencies without causing a gn check error